app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/CommonTriageQuestionsController.scala

```diff
@@ -560,17 +560,24 @@ class CommonTriageQuestionsController @Inject() (
             )
           )
 
-        val form     = sentSelfAssessment.fold(alreadySentSelfAssessmentForm)(alreadySentSelfAssessmentForm.fill)
-        val backLink = triageAnswers.fold(
-          _.fold(
-            _ => routes.MultipleDisposalsTriageController.whenWereContractsExchanged(),
-            _ => routes.MultipleDisposalsTriageController.checkYourAnswers()
-          ),
-          _.fold(
-            _ => routes.SingleDisposalsTriageController.whenWasDisposalDate(),
-            _ => routes.SingleDisposalsTriageController.checkYourAnswers()
+        val isAmendReturn: Boolean               = state.fold(_ => false, _.isAmendReturn)
+        val originalSubmissionId: Option[String] =
+          state.toOption.flatMap(_.amendReturnData.map(_.originalReturn.summary.submissionId))
+        val form                                 = sentSelfAssessment.fold(alreadySentSelfAssessmentForm)(alreadySentSelfAssessmentForm.fill)
+        val backLink                             = if (isAmendReturn && sentSelfAssessment.isEmpty) {
+          controllers.accounts.homepage.routes.HomePageController.viewSentReturn(originalSubmissionId.getOrElse(""))
+        } else {
+          triageAnswers.fold(
+            _.fold(
+              _ => routes.MultipleDisposalsTriageController.whenWereContractsExchanged(),
+              _ => routes.MultipleDisposalsTriageController.checkYourAnswers()
+            ),
+            _.fold(
+              _ => routes.SingleDisposalsTriageController.whenWasDisposalDate(),
+              _ => routes.SingleDisposalsTriageController.checkYourAnswers()
+            )
           )
-        )
+        }
```

Comment:
I feel its good to avoid `.getOrElse("")` calls with this way

`        val backLink                             =
          (isAmendReturn, sentSelfAssessment.isEmpty, originalSubmissionId) match {
            case (true, true, Some(submissionId)) =>
              controllers.accounts.homepage.routes.HomePageController.viewSentReturn(submissionId)
            case _ =>
              triageAnswers.fold(
                _.fold(
                  _ => routes.MultipleDisposalsTriageController.whenWereContractsExchanged(),
                  _ => routes.MultipleDisposalsTriageController.checkYourAnswers()
                ),
                _.fold(
                  _ => routes.SingleDisposalsTriageController.whenWasDisposalDate(),
                  _ => routes.SingleDisposalsTriageController.checkYourAnswers()
                )
              )
        }`

Reply:
Yeah, this makes code look better. Done

Reply:
Looks good now